workspaces/spectacle/src/spectacle.ts

```diff
@@ -232,6 +237,51 @@ export async function makeSpectacle(opticContext: IOpticContext) {
         );
         return Promise.resolve(endpointChanges);
       },
+      field: (
+        parent: any,
+        { fieldId }: { fieldId: string },
+        context: GraphQLContext
+      ) => {
+        const fieldNodeWrapper = context
+          .spectacleContext()
+          .shapeQueries.findNodeById(fieldId);
+
+        if (
+          !fieldNodeWrapper ||
+          fieldNodeWrapper.result.type !== shapes.NodeType.Field
+        )
+          return Promise.resolve(null);
+
+        let fieldNode = fieldNodeWrapper.result;
+
+        let shapeNodeWrapper = context
+          .spectacleContext()
+          .shapeQueries.listIncomingNeighborsByType(
+            fieldId,
+            shapes.NodeType.Shape
+          )
+          .results.find((shapeNode) => {
+            return shapeNode.result.type === shapes.NodeType.Shape;
+          });
```

Comment:
I _think_ I've updated the typing such that `listIncomingNeighborsByType` correctly returns the right type, so you only don't need to find (just check length >=1) - similar to https://github.com/opticdev/optic/blob/develop/workspaces/graph-lib/src/endpoints-graph/index.ts#L333-L341

You could also add this query (shapeFromField) in the FieldNodeWrapper class in graph-lib

Reply:
You're right, it does return the more specific class, but checking for length and returning an array index breaks my nice functional chain. If anything, `NodeListWrapper` needs a `first` method hiding all that nastiness, making up for `Array`s lack of one. 

Feel free to contribute a clean up for this, but I'm going to pass on it for today (might pick up in the morning!).

Reply:
Turns out it still returns the generic `shapes.Node`, with further assertion being necessary to limit it down to a `shapes.ShapeNode`. I attempted a version where `NodeListWrapper` is a class with a `first` method, but that didn't really allow for any tangible improvements. It wasn't enough of a direction to leave it, only muddling up the water further. The typing to this graphing definitely needs some love, but that's outside of the scope of this PR.